core/src/com/put/urbantraffic/CityGenerator.java

```diff
@@ -0,0 +1,118 @@
+
+import java.util.concurrent.ThreadLocalRandom;
+
+public class CityGenerator {
+    private final int width = 2 * 16;
+    private final int height = 2 * 9;
+    private final int crossingAmount = 3;
+
+    int[][] generate() {
+        val grid = new int[height][width];
+        val heightOffset = (height + 1) % 2;
+        for (int i = 0; i < width; i++) {
+            grid[height / 2 - heightOffset][i] = 1;
+        }
+        for (int i = 0; i < height; i++) {
+            grid[i][width / 2] = 1;
+        }
+        for (int iter = 0; iter < crossingAmount; iter++) {
+            int x = width / 2;
+            int y = heightOffset / 2;
+            while (grid[y][x] == 9 || grid[y][x] == 1) {
+                x = 2 * generateRandomInt(1, (width - 2) / 2);
+                y = 2 * generateRandomInt(1, (height - 2) / 2);
+            }
+            grid[y][x] = 9;
+            // right
+            for (int i = x + 1; i < width; i++) {
+                if (grid[y][i] == 1 || grid[y][i] == 9) {
+                    fillArray(grid[y], x + 1, i, 1);
+                    grid[y][i] = 9;
+                    break;
+                }
+            }
+            // left
+            for (int i = x - 1; i > -1; i--) {
+                if (grid[y][i] == 1 || grid[y][i] == 9) {
+                    fillArray(grid[y], i + 1, x, 1);
+                    grid[y][i] = 9;
+                    break;
+                }
+            }
+            // down
+            for (int i = y + 1; i < height; i++) {
+                //noinspection DuplicatedCode
+                if (grid[i][x] == 1 || grid[i][x] == 9) {
+                    for (int j = y + 1; j < i; j++) {
+                        grid[j][x] = 1;
+                    }
+                    grid[i][x] = 9;
+                    break;
+                }
+            }
+            // up
+            for (int i = y - 1; i > -1; i--) {
+                //noinspection DuplicatedCode
+                if (grid[i][x] == 1 || grid[i][x] == 9) {
+                    for (int j = i + 1; j < y; j++) {
+                        grid[j][x] = 1;
+                    }
+                    grid[i][x] = 9;
+                    break;
+                }
+            }
+        }
+
+        for (int i = 0; i < height; i++) {
+            for (int j = 0; j < width; j++) {
+                if (grid[i][j] == 9) {
+                    grid[i][j] = checkIfShouldStayCrossing(grid, i, j);
+                }
+            }
+        }
+        return grid;
+    }
+
+    private int checkIfShouldStayCrossing(int[][] grid, int i, int j) {
+        val sumOfNeighbours = grid[i + 1][j] + grid[i - 1][j] + grid[i][j + 1] + grid[i][j - 1];
+        if (sumOfNeighbours == 2) {
+            return 8;
+        }
+        return 9;
+    }
+
+    /**
+     * @param min - minimum value
+     * @param max - maximum value
+     * @return random int number from range [min, max]
+     */
+    private int generateRandomInt(int min, int max) {
```

Comment:
Big chance we will use it later. Maybe I should move it to some `RandomUtils` file